### ISSUE TITLE
Memory leaks in MonoLoader?

### ISSUE THREAD

**Issue by jamiebullock** [MTG/essentia#634]
I have some memory leaks in my app, which I can trace back (with Valgrind) to the following:

```
==1== 2,028 (1,440 direct, 588 indirect) bytes in 3 blocks are definitely lost in loss record 171 of 251
==1==    at 0x4C2E0EF: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==1==    by 0x5219929: essentia::streaming::connect(essentia::streaming::SourceBase&, essentia::streaming::DevNullConnector) (in /usr/local/lib/libessentia.so)
==1==    by 0x53F667A: essentia::streaming::MonoLoader::MonoLoader() (in /usr/local/lib/libessentia.so)
==1==    by 0x52AA0EA: essentia::EssentiaFactory<essentia::streaming::Algorithm>::Registrar<essentia::streaming::MonoLoader, essentia::standard::MonoLoader>::create() (in /usr/local/lib/libessentia.so)
==1==    by 0x5217468: essentia::EssentiaFactory<essentia::streaming::Algorithm>::create_i(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const (in /usr/local/lib/libessentia.so)
==1==    by 0x53F6A0A: essentia::standard::MonoLoader::createInnerNetwork() (in /usr/local/lib/libessentia.so)
==1==    by 0x52C4A39: essentia::EssentiaFactory<essentia::standard::Algorithm>::Registrar<essentia::standard::MonoLoader, essentia::standard::MonoLoader>::create() (in /usr/local/lib/libessentia.so)
==1==    by 0x4AE54F: essentia::EssentiaFactory<essentia::standard::Algorithm>::create_i(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, essentia::Parameter const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, essentia::Parameter const&) const (in /usr/local/bin/extractron)
==1==    by 0x4AADE1: essentia::EssentiaFactory<essentia::standard::Algorithm>::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, essentia::Parameter const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, essentia::Parameter const&) (algorithmfactory.h:170)
```

I am freeing the object returned by `AlgorithmFactory::create()` and even calling `essential::shutdown()` yet certain memory is not freed.

Looking briefly at the [source code](https://github.com/MTG/essentia/blob/master/src/algorithms/io/monoloader.cpp) for MonoLoader Ii can see a couple of `new` but no `delete`...

**Comment by jamiebullock** [comment]
I should add that I am using the [Essentia Docker image](https://hub.docker.com/r/mtgupf/essentia/) although I suspect the issue manifests elsewhere

**Comment by sansinghsanjay** [comment]
I am also facing same issue. When I am using MonoLoader and AudioLoader in Python, I am getting "Segmentation Fault (Core Dumped)" .

**Comment by audiofeature** [comment]
I can indirectly confirm that a memory leak still exists in v2.1_beta4 release, as I am starting multiple Python processes using Essentia with MonoLoader simulateneously, working on large batches of audio files and the memory consumption of all processes increases steadily, until the point where some of them get killed with out of memory after a few hours.

**Comment by dbogdanov** [comment]
There was a fix done in 5ea5b83371b3c4378c6b355c9d35b57e947256ab after the v2.1_beta4 release. Does the problem still persist?
